applications/ShallowWaterApplication/custom_utilities/sw_math_utils.h

```diff
@@ -0,0 +1,129 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_SW_MATH_UTILS_H_INCLUDED
+#define KRATOS_SW_MATH_UTILS_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SwMathUtils
+ * @ingroup ShallowWaterApplication
+ * @brief Various mathematical utilities functions
+ * @author Miguel Maso Sotomayor
+ */
+template<class TDataType>
+class SwMathUtils
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef std::size_t IndexType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TMatrixTypeA, class TMatrixTypeB>
+    static void AddMatrix(
```

Comment:
This is generic enough to be in the Core math utils

Reply:
I think the same, just now I'm preparing another PR to move it to the core